src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -289,6 +292,12 @@ export const GlobalProvider = memo(
             }, 100);
             return () => clearTimeout(timerId);
         }, [nodeChanges, edgeChanges]);
+        useEffect(() => {
+            const timerId = setTimeout(() => {
+                sessionStorage.setItem('cachedViewport', JSON.stringify(getViewport()));
+            }, 100);
+            return () => clearTimeout(timerId);
+        }, [vpx, vpy, vpZoom]);
```

Comment:
Because I want it to update when these values change, not when the reference to the function changes. And idk if just calling getViewport() in this would even work properly

Reply:
Though, I never actually tried just using getViewport, so maybe that just works. I'll try that out when I have time

Reply:
No, I think you're right. It would just be good to have a nicer interface. The `v*` variables are only used here. Maybe we could just do `Object.values(getViewport())` for the deps array?

Reply:
Also, could you remove `sessionStorage.setItem('cachedViewport', JSON.stringify(getViewport()));` from the above effect that handles nodes?

Reply:
> No, I think you're right. It would just be good to have a nicer interface. The `v*` variables are only used here. Maybe we could just do `Object.values(getViewport())` for the deps array?

I haven't tested it, but I don't think memoization works like that. I'm pretty sure doing that would cause it to re-compute every render. If I remember correctly, react memoization works via value references rather than strict value comparison, meaning if you're getting new references (regardless of if the numbers themselves are the same), then every time it does a comparison it will still recompute. That might be wrong though, I'd have to go look at the docs again

Reply:
No, you're right, it works via reference comparisons. I wanted to suggest `useEffect(..., Object.values(getViewport()))`, not `useEffect(..., [Object.values(getViewport())])`. It's the same trick I use for a few `useMemo`s.

Reply:
That actually doesn't work. Just tried it.

Reply:
That actually doesn't work. Just tried it.

Reply:
That actually makes me a bit concerned for the other useMemos we use that trick for. Are you 100% sure its a valid thing to do?

Reply:
I use it everywhere and it worked so far. What exactly did you try?